pkg/docker/config/config.go

```diff
@@ -232,7 +249,7 @@ func modifyJSON(sys *types.SystemContext, editor func(auths *dockerConfigFile) (
 		}
 	}
 
-	auths, err := readJSONFile(path, false)
+	auths, err := readJSONFile(path, legacyFormat)
```

Comment:
This won’t work; `readJSONFile` would convert the data from the legacy format, and the code below would write new-format data to the legacy-format path.

I don’t at all think we need to add write support for the legacy format here.

Either `getPathToAuth` could get an `acceptLegacy` (`readOnly`?) parameter, or `LegacyFormatAuthFilePath` could be handled only in the read path, in `GetAuthentication` changing the path used by `homeDirDockerCFG`.

Reply:
or how about throw an error if this is called with a SystemContext that has `LegacyFormatAuthFilePath` set ?

Reply:
… at this precise point, `if legacyFormat` (assuming `AuthFilePath` is documented to take precedence over `LegacyFormatAuthFilePath`, and we won’t pedantically throw an error every time both are specified).

Yes, that would work nicely as well, and allow us to add support for editing the legacy format in the future if it ever became necessary.